File: workoutFront/src/apis/response/user/get-sign-in-user.response.dto.ts
// import { ResponseCode } from '@types/enum';
import { ResponseCode } from '../../../types/enum';
import ResponseDto from '../response.dto';
// import { User } from '@types/interface';
import User from '../../../types/interface/user.interface';


export default interface GetSignInUserResponseDto extends ResponseDto, User{
    //ResponseDto, User을 extends 하였으므로 아래 모든 코드가 필요없다. ResponseDto가 code, message를 User가 message, email, profileImage를 가지고 있음
    // "code": ResponseCode;
    // "message": string;
    // "email": string;
    // "nickname": string;
    // "profileImage": string;
}

File: workoutbackend/src/main/java/com/flexy/workoutbackend/dto/response/user/GetSignInUserResponseDto.java
package com.flexy.workoutbackend.dto.response.user;

import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;

import com.flexy.workoutbackend.common.ResponseCode;
import com.flexy.workoutbackend.common.ResponseMessage;
import com.flexy.workoutbackend.dto.response.ResponseDto;
import com.flexy.workoutbackend.entity.UserEntity;

import lombok.Getter;

// 로그인 유저 정보 불러오기(GetSignInUser) API구현(35강)
@Getter
public class GetSignInUserResponseDto extends ResponseDto {

    private String id;
    private String realName;
    private String profileImage;

    private GetSignInUserResponseDto(UserEntity userEntity) {
        super(ResponseCode.SUCCESS, ResponseMessage.SUCCESS);
        this.id = userEntity.getUserId();
        this.realName = userEntity.getRealName();
        this.profileImage = userEntity.getProfileImage();
    }

    // 로그인 유저정보에 대한 요청은 크게 성공과 실패다. 코드는 아래와 같고 실패인 경우는 클라이언트로 보내는 ResponseEntity의
    // status를 UNAUTHORIZED로 하는것에 유의
    public static ResponseEntity<ResponseDto> success(UserEntity userEntity) {
        GetSignInUserResponseDto result = new GetSignInUserResponseDto(userEntity);
        return ResponseEntity.status(HttpStatus.OK).body(result);
    }

    public static ResponseEntity<ResponseDto> notExistedUser() {
        ResponseDto result = new ResponseDto(ResponseCode.NOT_EXISTED_USER, ResponseMessage.NOT_EXISTED_USER);
        return ResponseEntity.status(HttpStatus.UNAUTHORIZED).body(result);
    }
}
